cmd/util/ledger/migrations/ordered_map_migration.go

```diff
@@ -0,0 +1,272 @@
+	panic("unreachable")
+}
+
+func (v RawStorable) Clone(_ *interpreter.Interpreter) interpreter.Value {
+	panic("unreachable")
+}
+
+func (RawStorable) DeepRemove(_ *interpreter.Interpreter) {
+	// NO-OP
+}
+
+func (r RawStorable) Encode(enc *atree.Encoder) error {
+	return enc.CBOR.EncodeRawBytes(r)
+}
+
+func (r RawStorable) ByteSize() uint32 {
+	return uint32(len(r))
+}
+
+func (r RawStorable) StoredValue(storage atree.SlabStorage) (atree.Value, error) {
+	return r, nil
+}
+
+func (r RawStorable) ChildStorables() []atree.Storable {
+	return nil
+}
+
+func (r RawStorable) Storable(_ atree.SlabStorage, _ atree.Address, _ uint64) (atree.Storable, error) {
+	return r, nil
+}
+
+func splitPayloads(inp []ledger.Payload) (fvmPayloads []ledger.Payload, storagePayloads []ledger.Payload, slabPayloads []ledger.Payload) {
+	for _, p := range inp {
+		if state.IsFVMStateKey(
+			string(p.Key.KeyParts[0].Value),
+			string(p.Key.KeyParts[1].Value),
+			string(p.Key.KeyParts[2].Value),
+		) {
+			fvmPayloads = append(fvmPayloads, p)
+			continue
+		}
+		if bytes.HasPrefix(p.Key.KeyParts[2].Value, []byte(atree.LedgerBaseStorageSlabPrefix)) {
+			slabPayloads = append(slabPayloads, p)
+			continue
+		}
+		// otherwise this is a storage payload
+		storagePayloads = append(storagePayloads, p)
+	}
+	return
+}
+
+func (m *OrderedMapMigration) initialize(payload []ledger.Payload) ([]ledger.Payload, error) {
+	fvmPayloads, storagePayloads, slabPayloads := splitPayloads(payload)
+
+	m.ledgerView = NewView(append(fvmPayloads, slabPayloads...))
+	m.initPersistentSlabStorage(m.ledgerView)
+	m.initIntepreter()
+	return storagePayloads, nil
+}
+
+func (m *OrderedMapMigration) migrate(storagePayloads []ledger.Payload) ([]ledger.Payload, error) {
+	groupedByOwnerAndDomain := make(map[string](map[string][]Pair))
+
+	for _, p := range storagePayloads {
+		owner, entry :=
+			string(p.Key.KeyParts[0].Value),
+			string(p.Key.KeyParts[2].Value)
+		// if the entry doesn't contain a separator, we just ignore it
+		// since it is storage metadata and does not need migration
+		if !strings.Contains(entry, "storage\x1f") &&
+			!strings.Contains(entry, "public\x1f") &&
+			!strings.Contains(entry, "contract\x1f") &&
+			!strings.Contains(entry, "private\x1f") {
+			m.Log.Info().Msgf("Ignoring key in storage payloads: %s", entry)
+			continue
+		}
+		splitEntry := strings.Split(entry, "\x1f")
+		domain, key := splitEntry[0], splitEntry[1]
+		value := p.Value
+
+		domainMap, domainOk := groupedByOwnerAndDomain[owner]
+		if !domainOk {
+			domainMap = make(map[string][]Pair)
+		}
+		keyValuePairs, orderedOk := domainMap[domain]
+		if !orderedOk {
+			keyValuePairs = make([]Pair, 0)
+		}
+		domainMap[domain] = append(keyValuePairs, Pair{Key: key, Value: value})
+		groupedByOwnerAndDomain[owner] = domainMap
+	}
+
+	for owner, domainMaps := range groupedByOwnerAndDomain {
+		for domain, keyValuePairs := range domainMaps {
+			address, err := common.BytesToAddress([]byte(owner))
```

Comment:
we had some issues in the past with addresses with leading zeros, I don't remember the details on what was the issue (maybe using Bytes() method), just want to make sure we double check if this one is safe cc: @SupunS